spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -65,20 +70,22 @@ public String send(final String topic, Message message) throws RuntimeException
 			try {
 				return Publisher.defaultBuilder(TopicName.create(projectId, topic))
 						.setExecutorProvider(executorProvider).build();
-			}
-			catch (IOException e) {
+			} catch (IOException e) {
 				e.printStackTrace();
 			}
 			return null;
 		});
 
-		try {
-			return publisher.publish(PubsubMessage.newBuilder()
-					.setData(ByteString
-							.copyFrom(message.getPayload().toString().getBytes()))
-					.build()).get();
+		Object pubsubMessageObject = messageConverter.fromMessage(message, PubsubMessage.class);
+
+		if (!(pubsubMessageObject instanceof PubsubMessage)) {
+			throw new MessageConversionException("The specified converter must produce"
+					+ "PubsubMessages to send to Google Cloud Pub/Sub.");
 		}
-		catch (Exception e) {
+
+		try {
+			return publisher.publish((PubsubMessage) pubsubMessageObject).get();
```

Comment:
What is this `.get()` about?
If we talk about something like `Future<?>`, then let's return that exactly!
Or even better `ListenableFuture`.
The end-user will make decision to block the result or handle asynchronously.

The good sample for that is the same `KafkaTemplate.send()`